### DISCUSSION TITLE
Has anyone had luck with Kodi?

### DISCUSSION THREAD

**Discussion by kogren161** [tgorgdotcom/locast2plex#174] [Q&A]
I have gotten Plex and Jellyfin to work, I was trying to see if I could make Kodi work, but seem to be having trouble finding the settings for the IP for the HDhomerun PVR client.  I'm looking for a roku client that operates smoothly.  I really like the commercial skip function of Plex, but find the live tv fastforward and rewind to be clunky.

**Comment by libersongm** [comment]
I was going to look into Jellyfin, if you had any tips.  I would install on Ubuntu Budgie 20.04 I would take them.

Thanks in advance.

Very Respectfully,
Gary

**Comment by kogren161** [comment]
I am working from within windows, I have played a bit with making this all work in Linux Mint 2.0 but have not had luck as to yet.  My most difficult issue with Jellyfin in windows was creating the XMLTV file for the guide.  This video was most helpful https://www.youtube.com/watch?v=eAsIugiNbbc.  I am new to most of this top to bottom, but anything I can do to help I will.  I like jellyfin for the most part.

**Reply by libersongm** [reply]
Thanks so much, the XMLTV file is my challenge as well.   Followed a bunch of videos but no luck.   Will try the one you said and keep you posted.   Like to tinker so will see how it goes.  Thanks again and good luck with Kodi.

Very Respectfully,
Gary

**Reply by libersongm** [reply]
Sorry stupid question had watched that video but had issues as it is no longer supported.  Did you use the update one?  Thanks.

**Reply by kogren161** [reply]
I was getting a 404 error for the http://zap2xml.awardspace.info/ link.  I got it here https://www.reddit.com/r/Cordcutting/comments/ic0fmc/zap2xml_anybody_have_the_install_file/

**Reply by kogren161** [reply]
Oh, also I was not able to make it work following this fellow's pearl instructions, I used the .exe and I put it in a .bat file to run it as a task in windows.  I am not familiar with Pearl, so that was likely the problem.

**Reply by libersongm** [reply]
Thanks am an old perl hack so will see.  I couldn't find the original script.  Thanks again.  Let you know how it goes.   Good luck with Kodi!

Very Respectfully,
Gary

**Reply by libersongm** [reply]
Sorry one last questions anything special to login to tvguide?  Get unknown token error.

Thanks,
Gary

**Reply by kogren161** [reply]
I was unsure at first if I needed to replace the asterisks for the username and password, but I chose to and that worked.  I also changed the folder names, and the path.  I was not sure what the -z -F -O, and -T represented.  Mine is using the .exe

zap2xml.exe -z -u USERNAME -p PASSWORD-o D:\Downloads\JellyfinGuideData\tv_grab_file.xmltv -c D:\Downloads\JellyfinGuideData\cache2 -F -O -T

**Reply by libersongm** [reply]
Thanks.  Did same but have found a thread with same error I am getting token error.   Will keep at it thanks.

**Reply by kogren161** [reply]
You bet.  best of luck.

**Reply by libersongm** [reply]
I got mc2xml to work.   They have a windows version.   http://mc2xml.awardspace.info/

Thanks for your help.

Very Respectfully,
Gary

**Reply by kogren161** [reply]
I am beginning to play with getting Jellyfin up in Linux.  The problem I am having is getting the XMLTV file to populate the guide.  Running the perl script works, and generates the file, when I point Jellyfin to it, it doesn't seem to provide any information.  I tried using the XMLTV file that was working for Jellyfin in Windows, but, no dice.

When I refresh the guide data in Windows it takes a minute, but in Linux it show the first bit of blue, then finishes within a couple of seconds.

**Reply by libersongm** [reply]
I used mc2xml on Linux using the following

US: mc2xml -2 -c us -g 12345

Where 12345 is your ZIP code.  It populates the guide fine

Gary

**Reply by kogren161** [reply]
Thank you for the suggestion.  It seems maybe it is a permissions thing, I am getting the same result using this method.  I tried giving permission to the xmltv file to Jellyfin, but I get the same result.

**Reply by ElPedroLRamos** [reply]
Can confirm I got mc2xml to work on Windows to Jellyfin.

**Reply by kogren161** [reply]
Thanks for letting me know.  My setup is working well in Windows.  I can't seem to get Linux Jellyfin to read the XMLTV file.  Not the one I generate in Windows, nor the one I generate in Linux.  BTW I am running Linux Mint 2.0

I have gotten Plex to run in both Windows and Linux, but the two servers seem to generate a conflict.  Each installation recognizes that it cannot see the other server even when they are given the same name.
